protocol-units/bridge/service/src/lib.rs

```diff
@@ -70,15 +76,27 @@ where
 									ChainId::ONE => {
 										let fut = process_action(action, one_client.clone());
 										if let Some(fut) = fut {
-											let jh = tokio::spawn(fut);
+											let jh = tokio::spawn({
+												let client_lock_clone = one_client_lock.clone();
+												async move {
+													let _lock = client_lock_clone.lock().await;
+													fut.await
```

Comment:
Do you think any error handling could potentially be useful here, like

```
if let Err(e) = fut.await {
    eprintln!("Error in task: {:?}", e);
}
```

likewise in line 97?

Reply:
The error is managed later. The fut return is added to the list of future (line 86) to execute and after execution the error is handle. See line 145 for example.